chainer/functions/connection/deconvolution_nd.py

```diff
@@ -0,0 +1,356 @@
+        if _cudnn_version >= 4000:
+            algo = libcudnn.getConvolutionBackwardFilterAlgorithm(
+                handle, gy_desc.value, gx_desc.value,
+                self.conv_desc.value, self.filter_desc.value,
+                _bwd_filter_pref, workspace_size)
+
+            libcudnn.convolutionBackwardFilter_v3(
+                handle, one.data, gy_desc.value, gy.data.ptr,
+                gx_desc.value, x.data.ptr, self.conv_desc.value,
+                algo, workspace.data.ptr, workspace_size,
+                zero.data, self.filter_desc.value, gW.data.ptr)
+        else:
+            libcudnn.convolutionBackwardFilter_v2(
+                handle, one.data, gy_desc.value, gy.data.ptr,
+                gx_desc.value, x.data.ptr, self.conv_desc.value,
+                zero.data, self.filter_desc.value, gW.data.ptr)
+
+        if b is None:
+            return gx, gW
+        else:
+            return gx, gW, gb
+
+    def backward(self, inputs, grad_outputs):
+        x, W = inputs[:2]
+        b = inputs[2] if len(inputs) == 3 else None
+        gy = grad_outputs[0]
+
+        xp = cuda.get_array_module(*inputs)
+        if xp is numpy:
+            return self._backward_xp(x, W, b, gy, numpy)
+        elif self._use_cudnn(x, W):
+            return self._backward_cudnn(x, W, b, gy)
+        else:
+            return self._backward_xp(x, W, b, gy, cuda.cupy)
+
+
+def deconvolution_nd(x, W, b=None, stride=1, pad=0, outsize=None,
+                     use_cudnn=True):
+    """N-dimensional deconvolution function.
+
+    This is an implementation of N-dimensional deconvolution which generalizes
+    two-dimensional one. It takes three variables: input ``x``, the filter
+    weight ``W``, and the bias vector ``b``.
+
+    Args:
+        x (~chainer.Variable or :class:`numpy.ndarray` or cupy.ndarray):
+            Input data of shape :math:`(n, c_I, d_1, d_2, ..., d_N)`.
+        W (~chainer.Variable or :class:`numpy.ndarray` or cupy.ndarray):
+            Weight data of shape :math:`(c_I, c_O, k_1, k_2, ..., k_N)`.
+        b (~chainer.Variable or :class:`numpy.ndarray` or cupy.ndarray):
+            Bias vector of length :math:`c_O` (optional).
+        stride (int or tuple of ints): Stride of filter applications
+            :math:`(s_1, s_2, ..., s_N)`. ``stride=s`` is equivalent to
+            ``(s, s, ..., s)``.
+        pad (int or tuple of ints): Spatial padding size for input arrays
+            :math:`(p_1, p_2, ..., p_N)`. ``pad=p`` is equivalent to
+            ``(p, p, ..., p)``.
+        outsize (tuple of ints): Expected output size of deconvolutional
+            operation. It should be a tuple of ints
+            :math:`(out_1, out_2, ..., out_N)`. Default value is ``None`` and
+            the outsize is estimated by input size, stride and pad.
+        use_cudnn (bool): If ``True``, then this function uses cuDNN if
+            available. Deconvolution using cuDNN supports more than
```

Comment:
Note that cuDNN supports more than one-dimensional deconvolution operations only.